docs/DeveloperGuide.md

```diff
@@ -1,38 +1,527 @@
+1. `Menu#deleteItem` then invokes `DeleteItemValidation#validateCommand` which in turn, calls all of the following 
+validation method to check the arguments provided.
+    * `DeleteItemValidation#validateArgument` checks if the user input `String` is empty
+        * If the user input is empty, a message indicating that the input is empty is printed using `MenuUi#printError` 
+      and control is given back to `MoneyGoWhere`.
+    * `DeleteItemValidation#validateIndex` checks if the given index is valid and exists
+        * If the index is invalid, a message indicating the index does not exists is printed using `MenuUi#printError` 
+      and control is given back to `MoneyGoWhere`.
+1. `Menu#processDeleteItem` is then invoked, and it calls `Menu#removeItem` on the given index to delete it from the 
+list of items. Then, `Menu#save` is invoked to save the changes to the local storage file.
+1. `Router` object then calls `MenuUi#printCommandSuccess` to print a message indicating that the item has been 
+successfully added to the menu.
+
+
+##### Basic Mode [Delete an Item]()
+The expected inputs to add only one menu item into an order is as such:
+* `deleteitem`
+* `2`
+
+This sequence diagram models the interaction between various components in MoneyGoWhere when the user inputs the 
+command `deleteitem` or `2`.
+
+The general workflow of `deleteitem` is as follows:
+1. `MoneyGoWhere` then creates a new `Command` object using the user input, whose constructor invokes 
+`Parser#formatArguments` method to extract the arguments for each flag into a `Map`.
+1. `Router#handleRoute` is then invoked to process the command and calls `Router#assistRoute` which invokes 
+`MenuAssistant#deleteItem` method to run the `deleteitem` command.
+1. Once the command runs, it can be aborted at any time when the user inputs `/cancel`.
+1. `MenuAssistant#deleteItem` checks if the list of items is empty. If empty, a message indicating that there is 
+nothing to be deleted using `MenuUi#printError` and control is given back to `MoneyGoWhere`.
+1. `MenuAssistant#deleteItem` invokes `MenuAssistant#getIndex` to get the index of the item to be deleted.
+1. `MenuAssistant#getIndex` gets the index from the user and invokes `DeleteItemValidation#validateIndex` if the given 
+index is valid and exists.
+    * If the index is invalid, a message indicating the index does not exist is printed using `MenuUi#printError` 
+   and re-prompts the user to enter a new index.
+1. A new `Item` object is then created using the name and price given
+1. `Menu#removeItem` is then invoked on the given index to delete it from the list of items. Then, `Menu#save` is 
+invoked to save the changes to the local storage file.
+1. `Router#assistRoute` then calls `MenuAssistant#printResult` to print a message indicating that if the item has been 
+successfully added to the menu or if the user has cancelled the command accordingly.
+
+<hr>
+
+#### List all Items
```

Comment:
May be you want to include a sequence to explain this features because it is hard to understand?